ui/plugin-system/src/components/PluginEditor/PluginEditor.tsx

```diff
@@ -0,0 +1,46 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Box, FormControl, FormHelperText, InputLabel } from '@mui/material';
+import { PluginKindSelect } from '../PluginKindSelect';
+import { PluginSpecEditor } from '../PluginSpecEditor';
+import { PluginEditorProps, usePluginEditor } from './plugin-editor-api';
+
+/**
+ * A combination PluginKindSelect and PluginSpecEditor component, meant for editing the common pattern in our specs of
```

Comment:
maybe good to explain what the common pattern is here 😛

Reply:
Clearly the rest of that comment did not do a good enough job. 🤣 

> common pattern in our specs of a `plugin` property with a `Definition` object attached to it.

I'm basically trying to say "for editing the `plugin` property that's in our JSON a lot and points at a plugin definition". Any suggestions for how to word that better?

Reply:
I gave it a shot trying to provide some more detail in the comments. Hopefully it's a little clearer!